code/modules/mapfluff/ruins/spaceruin_code/hauntedtradingpost.dm

```diff
@@ -0,0 +1,362 @@
+		/datum/reagent/consumable/nutriment/vitamin = 5,
+		/datum/reagent/consumable/bungojuice = 1,
+		/datum/reagent/consumable/curry_powder = 1,
+		/datum/reagent/consumable/soymilk = 1,
+		/datum/reagent/consumable/tomatojuice = 1,
+		/datum/reagent/consumable/vitfro = 1,
+	)
+	icon_state = "ketchup"
+	fill_icon_thresholds = null
+
+// [Hazards & Traps]
+//cyborg holobarriers that die when the boss dies, how exciting
+/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield
+	desc = "A fragile holographic energy field projected by an AI core. It keeps unwanted humanoids at safe distance."
+
+/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield/Initialize(mapload)
+	. = ..()
+	GLOB.selfdestructs_when_boss_dies += src
+
+/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield/Destroy()
+	GLOB.selfdestructs_when_boss_dies -= src
+	return ..()
+
+//smes that produces power, until the boss dies then it self destructs and you gotta make your own power
+/obj/machinery/power/smes/magical/cybersun
+	name = "cybersun-brand power storage unit"
+	desc = "A high-capacity superconducting magnetic energy storage (SMES) unit. It looks like any other SMES unit, except this one says 'Cybersun' on it."
+	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
+	var/donk_ai_slave = FALSE
+
+/obj/machinery/power/smes/magical/cybersun/Initialize(mapload)
+	. = ..()
+	if(donk_ai_slave)
+		GLOB.selfdestructs_when_boss_dies += src
+
+/obj/machinery/power/smes/magical/cybersun/Destroy()
+	if(donk_ai_slave)
+		GLOB.selfdestructs_when_boss_dies -= src
+	return ..()
+
+//this is a trigger for traps involving doors and shutters
+//doors get closed and bolted, shutters get cycled open/closed
+/obj/machinery/button/door/invisible_tripwire
+	name = "Sonic Tripwire"
+	desc = "An invisible trigger for shutters and doors. Triggers when someone steps on the tile."
+	max_integrity = 50
+	invisibility = INVISIBILITY_ABSTRACT
+	anchored = TRUE
+	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
+	var/donk_ai_slave = FALSE
+	//can the trap trigger more than once?
+	var/multiuse = FALSE
+	//(if multiuse) how many times the trap can trigger. 0 or lower is infinite
+	var/uses_remaining = 0
+	//if true, the trap will unbolt all doors it bolted and cycle shutters a second time after a delay
+	var/resets_self = FALSE
+	//time before resets_self kicks in
+	var/reset_timer = 1.8 SECONDS
+	//when multiple tripwires are in the same suicide pact, they will all die when any of them die
+	var/suicide_pact = FALSE
+	//id of the suicide pact this tripwire is in
+	var/suicide_pact_id
+GLOBAL_LIST_EMPTY(tripwire_suicide_pact)
+
+/obj/machinery/button/door/invisible_tripwire/Initialize(mapload)
+	. = ..()
+	if(donk_ai_slave == TRUE)
+		GLOB.selfdestructs_when_boss_dies += src
+	if(suicide_pact == TRUE && suicide_pact_id != null)
+		GLOB.tripwire_suicide_pact += src
```

Comment:
I think you should look into the queuelink subsystem and see if you can make use of it instead of creating niche global lists.

Reply:
dont know what this is or what the advantages are

Reply:
It's a subsystem that links datums/atoms sharing a same id through signals. I'm telling you because in the long run it's better than stuff like code debt.